packages/react-test-renderer/src/ReactTestRendererScheduling.js

```diff
@@ -31,8 +31,39 @@ export function setNowImplementation(implementation: () => number): void {
   nowImplementation = implementation;
 }
 
-export function flushAll(): Array<mixed> {
-  yieldedValues = null;
+function verifyExpectedValues(expectedValues: Array<mixed>): void {
+  for (let i = 0; i < expectedValues.length; i++) {
+    const expectedValue = `"${(expectedValues[i]: any)}"`;
+    const yieldedValue =
+      i < yieldedValues.length ? `"${(yieldedValues[i]: any)}"` : 'nothing';
+    if (yieldedValue !== expectedValue) {
+      const error = new Error(
+        `Flush expected to yield ${(expectedValue: any)}, but ${(yieldedValue: any)} was yielded`,
+      );
+      // Attach expected and yielded arrays,
+      // So the caller could pretty print the diff (if desired).
+      (error: any).expectedValues = expectedValues;
+      (error: any).actualValues = yieldedValues;
+      throw error;
+    }
+  }
+
+  if (expectedValues.length !== yieldedValues.length) {
+    const error = new Error(
+      `Flush expected to yield ${expectedValues.length} values, but yielded ${
+        yieldedValues.length
+      }`,
+    );
+    // Attach expected and yielded arrays,
+    // So the caller could pretty print the diff (if desired).
```

Comment:
Can we please go ahead and implement this for the React repo? @cyan33 You want to do this?

Reply:
Of Course 👍